test/libsolidity/ASTJSON.cpp

```diff
@@ -254,10 +256,41 @@ BOOST_AUTO_TEST_CASE(documentation)
 	astJsonB = ASTJsonConverter(false, sourceIndices).toJson(c.ast("b"));
 	documentationB = astJsonB["nodes"][0]["documentation"];
 	BOOST_CHECK_EQUAL(documentationB, "This contract is empty and has a line-breaking comment.");
-
 }
 
 
+BOOST_AUTO_TEST_CASE(importAST)
+{
+	CompilerStack c;
+	c.addSource("a",
+		"contract C { function f(function() external payable returns (uint) x) "
+		"returns (function() external constant returns (uint)) {} }"
+	);
+	c.parseAndAnalyze();
+	c.compile();
+	map<string, unsigned> sourceIndices;
+	sourceIndices["a"] = 0;
+	Json::Value originalJson = ASTJsonConverter(false, sourceIndices).toJson(c.ast("a"));
+	string originalBinary = dev::test::bytecodeSansMetadata(c.object(c.contractNames()[0]).toHex());
+	// import the exported Json
+	map<string, Json::Value const*> sourceList;
+	sourceList["a"] = &originalJson;
+	c.reset(false);
+	assert(c.importASTs(sourceList));
+	//re-generate JSON and bytecode and compare
+	assert(c.analyze());
+	assert(c.compile());
+	Json::Value newJson = ASTJsonConverter(false, c.sourceIndices()).toJson(c.ast("a"));
+	assert(newJson == originalJson);
+	string newBinary = dev::test::bytecodeSansMetadata(c.object(c.contractNames()[0]).toHex());
+	// Note: Bytecode comparison will only succeed if the contracts don't import/inherit other contracts, because
+	// the other contract's bytecode will contain it's metadata, whose 'source'-field is usually
+	// generated from the solidity-code, but after importing will be generated from the Json.
+	// Therefore, in order to test more complicated contracts, the bytecodeSansMetadata-function needs to be extended to
+	// also remove metadatabytecode not only from the end but also from the middle of a given string.
+	assert(newBinary == originalBinary);
```

Comment:
`BOOST_CHECK_EQUAL` should be used instead of asserts here.

Reply:
Sorry I meant this for every `assert` in that function (there are a couple more above).

Reply:
oh, should have been obvious, thx. done now.